### PR TITLE
Using-API Week1 Vladimir

### PR DESCRIPTION
**Pull request by VladimirSeredinin**

### CODE REVIEW

File: 3-UsingAPIs/Week1/homework/ex1-johnWho.js [hunk 1]
```diff
@@ -8,18 +8,16 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
...
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
+      }
+      resolve(`${firstName} Doe`);
+    }, 1000);
+  });
 };
 
 function main() {
```
Comment (remarcmij) [inline 3-UsingAPIs/Week1/homework/ex1-johnWho.js:23]: Your `main()` function no longer works correctly. It no longer logs the results of the `getAnonName()` function because that function no longer takes a callback argument but instead returns a promise. You should adapt the `main()` function to use the promise and log the result or the error.

File: 3-UsingAPIs/Week1/homework/ex1-johnWho.js [hunk 2]
```diff
@@ -8,18 +8,16 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
...
 
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
```
Comment (remarcmij) [inline 3-UsingAPIs/Week1/homework/ex1-johnWho.js:16]: In this particular case, it does not make sense to call `resolve()` after you have called `reject()`.  It is better to _return early_. Note that calling `reject()` does not magically stop the calling function.

```suggestion
        reject(new Error("You didn't pass in a first name!"));
        return;
```

File: 3-UsingAPIs/Week1/homework/ex2-checkDoubleDigits.js [hunk 3]
```diff
@@ -12,8 +12,14 @@ Complete the function called `checkDoubleDigits` such that:
   "Expected a double digit number but got `number`", where `number` is the 
   number that was passed as an argument.
 ------------------------------------------------------------------------------*/
-function checkDoubleDigits(/* TODO add parameter(s) here */) {
-  // TODO complete this function
+function checkDoubleDigits(digit) {
+  return new Promise((resolve, reject) => {
+    if (digit > 9 && digit < 100) {
```
Comment (remarcmij) [inline 3-UsingAPIs/Week1/homework/ex2-checkDoubleDigits.js:17]:
```suggestion
    if (digit >= 10 && digit <= 99) {
```

Since we are talking in terms of double-digit numbers I would be inclined to express the bounds in the same terms. Just my personal preference, but your code is fine too, of course.

File: 3-UsingAPIs/Week1/homework/ex3-rollDie.js [hunk 4]
```diff
@@ -11,50 +11,50 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/3-Usin
...
-    }
-  });
+    })
+    .catch((error) => {
+      console.log(error.message);
+    });
 }
 
+/* At lines 26 and 27, the condition is written that if the roll is greater than 6, an error occurs (rolling off the table).*/
+
```
Comment (remarcmij) [inline 3-UsingAPIs/Week1/homework/ex3-rollDie.js:57]: Was this intended as the requested explanation as per the README?

- Does the problem described above still occur? If not, what would be your explanation? Add your answer as a comment to be bottom of the file.

If not, please add your explanation or otherwise try and improve on this one.

The code looks fine, though.

File: 3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js [hunk 5]
```diff
@@ -27,9 +27,12 @@ exercise file.
...
 
 function rollDice() {
-  // TODO Refactor this function
   const dice = [1, 2, 3, 4, 5];
-  return rollDie(1);
+  return Promise.all(
+    dice.map((item) => {
+      return rollDie(item);
+    })
+  );
```
Comment (remarcmij) [inline 3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:35]: You can shorten this to:

```suggestion
  return Promise.all(dice.map(rollDie));
```

Can you figure out why this would work too? Of course, your version is fine also.

File: 3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js [hunk 6]
```diff
@@ -27,9 +27,12 @@ exercise file.
...
 }
 
 function main() {
```
Comment (remarcmij) [inline 3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:38]: Can you add the explanation that was asked in the README?

- You may also notice that, in the case of a rejected promise, dice that have not yet finished their roll continue to do so. Can you explain why? Please add your answer as a comment to the end of the exercise file.

File: 3-UsingAPIs/Week1/homework/ex5-pokerDiceChain.js [hunk 7]
```diff
@@ -14,19 +14,15 @@ to expand the given promise chain to include five dice.
 // Do not change or remove it.
 const rollDie = require('../../helpers/pokerDiceRoller');
 
-function rollDice() {
+async function rollDice() {
```
Comment (remarcmij) [inline 3-UsingAPIs/Week1/homework/ex5-pokerDiceChain.js:17]: This week is not about `async`. That is next week's topic. Can you redo this exercise without using `async`?

**Review by remarcmij — CHANGES_REQUESTED** [review]
Hi @VladimirSeredinin, it looks like you have a good understanding of promises. There are however some issues in your homework that I would like you to fix. Please have a look at my comments below. Good luck!

File: 3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js (hunk 5 above)
Comment (VladimirSeredinin) [reply 3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:35]: Yeap, because rollDie accept only one parameter which is passed by map function

File: 3-UsingAPIs/Week1/homework/ex1-johnWho.js [hunk 8]
```diff
@@ -8,22 +8,23 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
...
+      resolve(`${firstName} Doe`);
+    }, 1000);
+  });
 };
 
 function main() {
-  getAnonName('John', console.log);
+  getAnonName('John')
+    .then((result) => console.log(result))
+    .catch((error) => console.log(error));
```
Comment (remarcmij) [inline 3-UsingAPIs/Week1/homework/ex1-johnWho.js:27]: Rather than printing the whole error object, better to just print the error message.

```suggestion
    .catch((error) => console.log(error.message));
```

**Review by remarcmij — APPROVED** [review]
Thanks for the updates, @VladimirSeredinin. I can now approve this PR.
